File: algorithms/53. Maximum Subarray/53. Maximum Subarray.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var sum = nums[0];
        
        for i in 1..<nums.count{
            sum += nums[i] + nums[i-1]
            sum = sum > nums[i] ? sum : nums[i]
        }// end for 
        return sum
    
    }// end func
}

File: algorithms/53. Maximum Subarray/53. Maximum Subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=nums[0]; 
        for(int i=1; i < nums.size(); i++){
            sum += nums[i] + nums[i-1];
            sum = sum > nums[i] ? sum : nums[i];
        }
        
        return sum;
    }// end func 
};
